### ISSUE TITLE
[Integration-Test] Add test for number of downloads.

### ISSUE THREAD

**Issue by jrbenny35** [mozilla/send#810]
Per testrail:

Send - Numbers of Downloads - 2 

1. Click the  "1 download" button and select the "2" option. | The "1 download" button changes into a "2 downloads" button.
2. Navigate to the Send homepage and observe the numbers of downloads for the uploaded file. | The 0/2 (0 downloads out of two) is correctly displayed.
3. Click the the uploaded file name. | You are redirected to the Share links page. The "2 downloads" button is correctly displayed.

We should add to the option to configure downloads to the ```share.py``` Page Object Model. The test should check 1. After 1 download, the remaining downloads left are 1 less than they were before. 2. The homepage correctly shows the amount of downloads left.

**Comment by prasanthp96** [comment]
Hi @jrbenny35 I can work on this issue.

**Comment by jrbenny35** [comment]
@prasanthp96 awesome!! Check out the docs found here: http://firefox-test-engineering.readthedocs.io/en/latest/

**Comment by AhmetCanSolak** [comment]
@jrbenny35 is this issue still available or merged already?

**Comment by jrbenny35** [comment]
@AhmetCanSolak it is still available however we have transitioned into using Javascript. Would you still be interested? If so, checkout the ```vnext``` branch, as the new tests will live there.

**Comment by dhyey35** [comment]
@jrbenny35 I have the project setup and in `vnext` branch. Can I work on this issue ?

**Comment by dhyey35** [comment]
@jrbenny35 while exploring the `vnext` branch I do not see a way of reproducing the issue i.e I cannot click on the file name to edit it as we can do in the production website.

**Comment by dhyey35** [comment]
@jrbenny35 I misunderstood first. Now If I correctly understand, I have to check if after 1 download does the 4 in below image change to 3.
![moz-1](https://user-images.githubusercontent.com/11143604/50688184-4f906c00-104a-11e9-858d-18f56f6a7aa3.png)

Am I understanding correctly ?

**Comment by jrbenny35** [comment]
> @jrbenny35 I misunderstood first. Now If I correctly understand, I have to check if after 1 download does the 4 in below image change to 3.
> 
> Am I understanding correctly ?

Yes, though this test should be at least 2 downloads. We Currently have a test for just 1 download.

**Comment by jrbenny35** [comment]
@dhyey35 yes they are but awesome! 😄 I'll take a look. You will have to open a PR against this repo and the ```vnext``` branch.

**Comment by dhyey35** [comment]
@jrbenny35 I was updating the docs (stopped that as I noticed you guys are updating to webdriver io v5) but noticed a weird thing, the test case I wrote is failing now because `getText` is now returning an `Object`, according to best of my knowledge it was previously returning `string`, now it throws `substring` is not a function error. I noticed that webdriverio has been updated from `4.14.1` to `4.14.2` by [this commit](https://github.com/mozilla/send/commit/339755e2920b38abfff29014d891ffd2e1a01fe1#diff-b9cfc7f2cdf78a7f4b91a753d10865a2). Can you please run the test and check at your end ? if it's an object probably I should update the code.

Also should I wait until you update to v5 or change the [documentation]( https://github.com/mozilla/send/tree/vnext/test/integration) now ?

**Comment by jrbenny35** [comment]
Hi @dhyey35. So I was thinking about updating to v5 earlier rather than later but there are still v4 docs and v5 still has some bugs. Maybe its not worth it at all for this project.

As regards the change, I can't find a change log [here](https://github.com/webdriverio-boneyard/v4/blob/master/CHANGELOG.md), so I wonder what the problem is. I get the same error as well. If you can find a fix that would be great. 

@dannycoates what do you think about documentation using webdriver v4? Should we try and update to 5 while the test suite is still small?

**Comment by dhyey35** [comment]
@jrbenny35 While debugging I found out that @dannycoates changed the selector from `.text-xs.text-grey-dark.w-full.mt-2.mb-2` to `send-archive > div` which returns `Copy URL`, Expiry Text and `Download` ( array of strings - that's why substring error is thrown ) text while `.text-xs.text-grey-dark.w-full.mt-2.mb-2` would just select the Expiry Text. But my code has issue too, if multiple files have been uploaded.

I am thinking about uploading a file called `txt-download-count-testfile.txt` and then looping through all filenames to find this particular file and then checking it's Expiry Text. 

Please let me know if you have any suggestions.

**Comment by jrbenny35** [comment]
A good suggestion based on what you said about multiple files would be adding to the ```download_page.js``` file a download list object, that had each uploaded file as a separate object. That way we could create an array or list of objects based on the amount of uploaded files we have. Each would have the properties specific to each upload.

Something like this: https://github.com/mozilla/notes/blob/master/test/integration/page_objects/list_page.js#L50, as I know javascript doesn't really do nested classes, but a class for each individual uploaded file card, and then we can do create a function to get and return the list. And feel free to add fixtures/whatever if they are needed for the test.

Also, this is why I want our download tests to run on CI but that is another story.
